src/util/aux_util.h

```diff
@@ -19,6 +19,9 @@ extern "C" {
 #define TPM2_ERROR_TEXT(r) "Error", "Code", r
 #define SIZE_OF_ARY(ary) (sizeof(ary) / sizeof(ary[0]))
 
+#define UNUSED(x) (void)(x)
+#define MAYBE_UNUSED __attribute__((unused))
```

Comment:
Do we really need MAYBE_UNUSED? I'm trying to figure out what this really does? Either the variable gets referenced or it doesn't. Sure their may be code paths that don't reference it, but that's normal.

Reply:
Also don't we want COMPILER_ATTR(unused)?

If loglevel=none is something that supposed to remove all logging code and disable it (ie no runtime level changes), shouldn't the macro be ifdef'd on the log level being none and the name be something like LOG_UNUSED() or something?

Reply:
I think what we want it is:
```C
#if MAXLOGLEVEL == LOGL_NONE
LOG_MAYBE_UNUSED COMPILER_ATTR(unused)
#else
LOG_MAYBE_UNUSED
#endif

This way we don't mark thinks always unused when they are MAYBE_UNUSED. We don't want to suppress legitimate unused warnings.

Reply:
MAYBE_UNUSED also applies to variables we use for assertions i.e. only in debug mode, mainly in test code.

Reply:
You can use `ifndef NDEBUG` from config.h for detecting debug builds. We just don't want a maybe macro that hides actual unused vars. I don't care what you call them.

Reply:
We just don't want a maybe macro that hides actual unused vars ... that we expect to have used paths.

Reply:
I can add some #ifdefs around MAYBE_UNUSED, no problem.
The main point of this PR is to catch early (i.e. in CI) PRs that break the TSS build with values for --with-loglevel flag different 
 that "warning" (the default) and/or --with-debug=no. Currently the master build fails when configured that way.
From the other hand having an unused variable marked and MAYBE_UNUSED doesn't cause any harm.

Reply:
It doesn't really cause any harm, except you can end up in conditions where their are extra variables floating around the code base; i'd like to avoid that as much as possible. Especially considering all the work we (mostly you) went through turning stuff on for code hygiene. Thanks.

Reply:
This: https://github.com/tpm2-software/tpm2-tss/pull/2044/commits/acc8b92890748a8b0e6d55e602f38017029ab630#diff-466b6f179d6c60bef3e5d528ebdc62bc54d25722b04f238d713fab2e8ff19515R31
should do the trick.